src/python/dxpy/nextflow/nextflow_builder.py

```diff
@@ -17,7 +17,17 @@
 parser = argparse.ArgumentParser(description="Uploads a DNAnexus App.")
 
 
-def build_pipeline_from_repository(repository, tag, profile="", git_creds=None, brief=False, destination=None, extra_args={}):
+def build_pipeline_with_npi(
+        repository=None,
+        tag=None,
+        cache_docker=False,
+        docker_secrets=None,
+        profile="",
+        git_creds=None,
+        brief=False,
+        destination=None,
+        extra_args=None
+):
     """
     :param repository: URL to a Git repository
```

Comment:
add docstring for new fields?

Reply:
[8fa5344](https://github.com/dnanexus/dx-toolkit/pull/1280/commits/8fa5344430a72b77a305f2d2f72f49b2ea7859e4)